src/parser/demonhunter/havoc/modules/talents/Demonic.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SpellLink from 'common/SpellLink';
+
+/**
+ * Example Report: https://www.warcraftlogs.com/reports/23dHWCrT18qhaJbz/#fight=1&source=16
+ */
+
+const META_BUFF_DURATION_EYEBEAM = 10000;
+
+class Demonic extends Analyzer{
+
+  eyeBeamCasts = 0;
+  goodDeathSweep = 0;
+  eyeBeamTimeStamp = undefined;
+  counter = undefined;
+  badCasts = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.DEMONIC_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EYE_BEAM), this.getTimeStamp);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.countCasts);
+  }
+
+  getTimeStamp(event) {
+    const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
+
+    if (hasMetaBuff) {
+      return;
+    }
+    this.eyeBeamCasts += 1;
+    this.eyeBeamTimeStamp = event.timestamp;
+
+    if (this.counter === undefined) {
+      this.counter = 0;
+      return;
+    }
+
+    if (this.counter < 2) {
+      this.badCasts += 1;
+    }
+
+    this.counter = 0;
+    console.log("reset to 0");
+  }
+
+  countCasts(event) {
+    if (this.eyeBeamTimeStamp !== undefined && (event.timestamp - this.eyeBeamTimeStamp) < META_BUFF_DURATION_EYEBEAM) {
+      this.goodDeathSweep += 1;
+      this.counter += 1;
+      console.log("counter #", this.counter, "  TS: ", this.eyeBeamTimeStamp );
+    }
+  }
+
+  get AvgPerEyeBeam(){
+    return this.goodDeathSweep / this.eyeBeamCasts;
+  }
+
+  get suggestionThresholds() {
+    return {
+      actual: this.badCasts,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: 'number',
+    };
+  }
+
+  suggestions(when) {
+    when(this.suggestionThresholds)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<>Squeezing in atleast 2 <SpellLink id={SPELLS.DEATH_SWEEP.id} /> during the meta phase you gain from <SpellLink id={SPELLS.EYE_BEAM.id} /> due to the <SpellLink id={SPELLS.DEMONIC_TALENT.id} /> talent is important to maximizing your dps.</>)
+          .icon(SPELLS.DEMONIC_TALENT.icon)
+          .actual(`${actual} bad casts`)
```

Comment:
if it's not too long, it would be better to replace `bad casts` with what actually happened. e.g. for Light of the Martyr I did:

```jsx
        .icon(SPELLS.LIGHT_OF_THE_MARTYR.icon)
        .actual(`${this.inefficientCasts.length} casts while Holy Shock was available`)
        .recommended(`No inefficient casts is recommended`);
```

This tells the reader what he did so he can more easily judge what to do with the info than if he's just being called bad.